app/js/notification.js

```diff
@@ -0,0 +1,124 @@
+module.exports = {};
+
+(function() {
+  "use strict";
+  let notifyId    = 0,
+      qNotifyArea = document.querySelector("#notification-container");
+
+  /**
+   * @private
+   * Generate a notification ID.
+   * @returns {Number}
+   */
+  function _id() {
+    notifyId++;
+    return notifyId;
+  }
+
+  /**
+   * Create the HTML for a notification.
+   * @see _make()
+   *
+   * @param {Object} notify - A notification object.
+   * @returns {String}
+   */
+  function _compile(notify) {
+    return `<div class="notification ${notify._class}" id="notification-${notify.id}">
+      <div class="notify-type">${notify.type}</div>
+      <div class="msg">${notify.msg}</div>
+    </div>`;
+  }
+
+  /**
+   * Create a notification object.
+   *
+   * @param {String} type - The type of notification we want to display.
+   * @param {String|Number} [msg=""] - The notification message.
+   * @returns {Object} A generated notification object.
+   */
+  function _make(type, msg) {
+    type = type.toLowerCase();
+    let details = {
+      id: _id(),
+      _class: type,
+      type: `${type.charAt(0).toUpperCase()}${type.substring(1)}`,
+      msg: msg || "",
+    };
+
+    details.html = _compile(details);
+    return details;
+  }
+
+  /**
+   * Hide a notification after a set time.
+   * @see _make
+   *
+   * @param {Object} notify - A notification object.
+   * @returns {Object} The notification object given.
+   */
+  function _hideDelayed(notify) {
```

Comment:
@le717 `_make` returns an object to `_show`, which returns an object to `_hideDelayed`.

My thinking is that the order that these functions appear in the code should reflect this order.